src/tess_asteroids/movingtpf.py

```diff
@@ -479,6 +487,173 @@ def _create_threshold_mask(
+        The function can also optionally return the x/y centroids and the ellipse
+        parameters (semi-major axis, A, semi-minor axis, B, and position angle, theta).
+        Pixels with distance <= R^2 from the pixel center to the asteroid position are
+        considered inside the aperture.
+        Ref: https://astromatic.github.io/sextractor/Position.html#ellipse-iso-def
+
+        Parameters
+        ----------
+        R: float
+            Value to scale the ellipse, the default is 3.0 which typically represents
+            well the isophotal limits of the object.
+        smooth: boolean
+            Whether to smooth the second-order moments by fitting a 3rd-order polynomial.
+            This helps to remove outliers and keep ellipse parameters more stable.
+        return_params: boolean
+            Return a ndarray with x/y centroids and ellipse parameters computed from
+            first- and second-order moments [X_cent, Y_cent, A, B, theta_deg].
+        plot: boolean
+            Generate a diagnostic plot with first- and second-order moments.
+
+        Returns
+        -------
+        aperture_mask: ndarray
+            Boolean 3D mask array with pixels within the ellipse.
+        ellipse_parameters: ndarray
+            If `return_params`, will return centroid and ellipse parameters
+            [X_cent, Y_cent, A, B, theta_deg] with shape (5, n_times).
+        """
+        # create a threshold mask to select pixels to use for moments
+        threshold_mask = self._create_threshold_mask(
+            threshold=3.0, reference_pixel="center"
+        )
+
+        X, Y, X2, Y2, XY = compute_moments(self.flux, threshold_mask)
+
+        if plot:
+            fig, ax = plt.subplots(2, 2, figsize=(9, 7))
+            fig.suptitle("Moments", y=0.94)
+            ax[0, 0].plot(
+                X + self.corner[:, 1], Y + self.corner[:, 0], label="Centroid"
+            )
+            ax[0, 0].plot(self.ephemeris[:, 1], self.ephemeris[:, 0], label="Ephem")
+            ax[0, 0].legend()
+            ax[0, 0].set_title("")
+            ax[0, 0].set_ylabel("Y")
+            ax[0, 0].set_xlabel("X")
+
+            ax[0, 1].plot(self.time, XY, c="tab:blue", lw=1, label="Moments")
+            ax[0, 1].set_ylabel("XY")
+            ax[0, 1].set_xlabel("Time")
+
+            ax[1, 0].plot(self.time, X2, c="tab:blue", lw=1)
+            ax[1, 0].set_ylabel("X2")
+            ax[1, 0].set_xlabel("Time")
+            ax[1, 1].plot(self.time, Y2, c="tab:blue", lw=1)
+            ax[1, 1].set_ylabel("Y2")
+            ax[1, 1].set_xlabel("Time")
+            if not smooth:
+                plt.show()
+
+        # fit a 3rd deg polynomial to smooth X2, Y2 and XY
+        # due to asteroid orbit projections, some tracks can show change in directions,
+        # a 3rd order polynomial can capture this.
+        if smooth:
+            # mask zeros and outliers
+            mask = ~np.logical_and(Y2 == 0, X2 == 0, XY == 0)
```

Comment:
```suggestion
            mask = ~np.logical_and(Y2 == 0, np.logical_and(X2 == 0, XY == 0))
```
np.logical_and only takes two conditions, so you have to nest them.